### PR TITLE
[12.0][FIX] hr_holidays_accrual_advanced: wizard crash

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py [hunk 1]
```diff
@@ -16,17 +16,14 @@ class HrLeaveAllocationAccrualCalculatorAccruement(models.TransientModel):
     )
     days_accrued = fields.Float(
         string='Number of Days',
-        readonly=True,
```
Comment (pedrobaeza) [inline hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: Why removing all these readonly?

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py [hunk 2]
```diff
@@ -83,17 +75,14 @@ def _recalculate(self):
...
                 'reason': accruement.reason,
-            }))
+            })
 
-        self.write({
-            'accrued': accrued,
-            'balance': balance,
-            'accruement_ids': accruement_ids,
-        })
+        self.accrued = accrued
```
Comment (pedrobaeza) [inline hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: Why not writing in block?

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py [hunk 3]
```diff
@@ -45,32 +42,27 @@ class HrLeaveAllocationAccrualBalanceCalculator(models.TransientModel):
...
+            'hr.leave.allocation.accrual.calculator.accruement'
+        ]
+        accruement_ids = CalculatorAccruement
 
         if not self.date:
-            self.write({
-                'balance': 0.0,
-                'accruement_ids': [(5, False, False)],
-            })
+            self.accrued = 0.0
```
Comment (pedrobaeza) [inline hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: Why not writing in block?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py (hunk 2 above)
Comment (alexey-pelykh) [reply hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: You can not write many2many relation on non-saved record without crash

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py (hunk 1 above)
Comment (alexey-pelykh) [reply hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: Probably I'll get those back

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py (hunk 3 above)
Comment (alexey-pelykh) [reply hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: You can not write many2many relation on non-saved record without crash

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py (hunk 2 above)
Comment (pedrobaeza) [reply hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: `accrued` are records? What a bad variable nomenclature then...

File: hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py [hunk 4]
```diff
@@ -60,17 +60,17 @@ class HrLeaveAllocationAccrualBalanceCalculator(models.TransientModel):
...
-    @api.multi
-    def _recalculate(self):
-        self.ensure_one()
+        CalculatorAccruement = self.env[
+            'hr.leave.allocation.accrual.calculator.accruement'
+        ]
+        accruement_ids = CalculatorAccruement
 
         if not self.date:
             self.write({
```
Comment (pedrobaeza) [inline hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py]: OK, now that you have removed the intermediate method I see the problem: you can't do a write on an onchange. You must do `.update` in replacement of.

**Comment by alexey-pelykh** [comment]
@pedrobaeza works as expected, thanks!

**Review by pedrobaeza — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@grindtildeath if you have a spare minute, please review this one

**Review by sswapnesh — APPROVED** [review]

**Pull request closed without merging**